packages/theming/src/utils/colors.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import type { ChromaStatic } from 'chroma-js';
+
+// @ts-expect-error Ignoring missing type definition
+import chroma from 'chroma-js/src/chroma';
+// @ts-expect-error Ignoring missing type definition
+import interpolate from 'chroma-js/src/generator/mix';
+// @ts-expect-error Ignoring missing type definition
+import valid from 'chroma-js/src/utils/valid';
+// @ts-expect-error Ignoring missing type definition
+import scale from 'chroma-js/src/generator/scale';
+
+import 'chroma-js/src/io/rgb';
+import 'chroma-js/src/io/hex';
+import 'chroma-js/src/io/lab';
+import 'chroma-js/src/interpolator/rgb';
+import 'chroma-js/src/ops/alpha';
+import PALETTE from '../elements/palette';
+
+chroma.interpolate = interpolate;
+chroma.valid = valid;
+chroma.scale = scale;
+
+interface ChromaLight {
+  scale: ChromaStatic['scale'];
+  valid: ChromaStatic['valid'];
+  interpolate: ChromaStatic['interpolate'];
+}
+
+const chromaLight = chroma as ChromaLight;
+
+export const isValidColor = chromaLight.valid;
+
+/**
+ * Generates a color scale with contrast ratios closely aligned to Garden's color palettes.
+ *
+ * @param {string} hue The color string to generate the scale from.
+ *
+ * @returns An offset-based color palette.
+ */
+export function getCustomColorScale(hue: string) {
```

Comment:
fwiw, I'm less interested in exposing another theming utility and more about containing the `chroma-js` import gnarliness to a dedicated location (under `types` or something :shrug:). For example, here's a compact version of a file that deals with the `valid` function...

```ts
/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import type { ChromaStatic } from 'chroma-js';
import { default as _valid } from 'chroma-js/src/utils/valid';
import 'chroma-js/src/io/hex';

export const valid = _valid as ChromaStatic['valid'];
```

Reply:
Cool 🙌 . Here's what I'd recommend...

- add a `chroma-js` folder under `theming/src/utils` that looks something like the screenshot below
- in `getColor` simply `import { scale, valid } from './chroma-js';`
- add the new `chroma-js.d.ts` under the project `/utils/build` with simple module declarations that clear the TS errors
- depending on design feedback, either leave `getColor` (nearly) as it is on `next`, or add a local `toScale` function similar to the other local functions that invokes `scale` and portions out the hue shades based on your target contrast ratios algorithm. I want to avoid the notion that Garden could provide a dedicated color scale utility outside of the `getColor` internals

<img width="896" alt="Screenshot 2024-04-30 at 12 22 38 PM" src="https://github.com/zendeskgarden/react-components/assets/143773/f947142a-e8ca-4ede-85c0-e9fc907018ef">